package/MDAnalysis/analysis/density.py

```diff
@@ -394,8 +395,16 @@ def _prepare(self):
                 logger.warning(msg)
             # Generate a copy of smin/smax from coords to later check if the
             # defined box might be too small for the selection
-            smin = np.min(coord, axis=0)
-            smax = np.max(coord, axis=0)
+            try:
```

Comment:
Just a thought here.

I can't think of a way to clean this up to have fewer lines of repeated code, however it might still be cleaner to move this part of the check to `_set_user_grid`?

Thoughts (tagging @hmacdope too). Ignore if you both don't think this is necessary.

Reply:
Hmm, this might be tricky.

Reply:
Okay, the way I think this can be done is to have a len(atomgroup) check done in _set_user_grid to see if there's an atom group first and then calculate smin and smax. We will also need to move the padding function into _set_user_grid as well. But tbh, I feel like this will only help with the abstraction of the script rather than how fast the code runs.

Reply:
IMO I think it can stay outside `set_user_grid` as it makes clear what is happening in main control flow for `_prepare`.